pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -0,0 +1,306 @@
+	}
+
+	log.Info("Windows VM has joined the cluster as a worker node", "ID", nc.ID())
+	return nil
+}
+
+// createUserDataSecret creates a secret 'windows-user-data' in 'openshift-machine-api'
+// namespace. This secret will be used to inject cloud provider user data for creating
+// windows machines
+func (r *ReconcileWindowsMachine) createUserDataSecret() error {
+	if r.signer == nil {
+		return errors.Errorf("failed to retrieve signer for private key: %v", wkl.PrivateKeyPath)
+	}
+
+	pubKeyBytes := ssh.MarshalAuthorizedKey(r.signer.PublicKey())
+	if pubKeyBytes == nil {
+		return errors.Errorf("failed to retrieve public key using signer for private key: %v", wkl.PrivateKeyPath)
+	}
+
+	// sshd service is started to create the default sshd_config file. This file is modified
+	// for enabling publicKey auth and the service is restarted for the changes to take effect.
+	userDataSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "windows-user-data",
+			Namespace: "openshift-machine-api",
+		},
+		Data: map[string][]byte{
+			"userData": []byte(`<powershell>
+			Add-WindowsCapability -Online -Name OpenSSH.Server~~~~0.0.1.0
+			$firewallRuleName = "ContainerLogsPort"
+			$containerLogsPort = "10250"
+			New-NetFirewallRule -DisplayName $firewallRuleName -Direction Inbound -Action Allow -Protocol TCP -LocalPort $containerLogsPort -EdgeTraversalPolicy Allow
+			Install-PackageProvider -Name NuGet -MinimumVersion 2.8.5.201 -Force
+			Install-Module -Force OpenSSHUtils
+			Set-Service -Name ssh-agent -StartupType ‘Automatic’
+			Set-Service -Name sshd -StartupType ‘Automatic’
+			Start-Service ssh-agent
+			Start-Service sshd
+			$pubKeyConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace '#PubkeyAuthentication yes','PubkeyAuthentication yes'
+			$pubKeyConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
+ 			$passwordConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace '#PasswordAuthentication yes','PasswordAuthentication yes'
+			$passwordConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
+			$authFileConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace 'AuthorizedKeysFile __PROGRAMDATA__/ssh/administrators_authorized_keys','#AuthorizedKeysFile __PROGRAMDATA__/ssh/administrators_authorized_keys'
+			$authFileConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
+			$pubKeyLocationConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace 'Match Group administrators','#Match Group administrators'
+			$pubKeyLocationConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
+			Restart-Service sshd
+			New-item -Path $env:USERPROFILE -Name .ssh -ItemType Directory -force
+			echo "` + string(pubKeyBytes[:]) + `"| Out-File $env:USERPROFILE\.ssh\authorized_keys -Encoding ascii
+			</powershell>
+			<persist>true</persist>`),
+		},
+	}
+
+	// check if the userDataSecret already exists
+	// TODO: Move this as first check, if this exist no need to create the vars unnecessarily
```

Comment:
Do we plan to address this TODO as a part of this PR or the watch userDataSecret story?